### ISSUE TITLE
edge function doesn't deploy when i switch to m2 chip

### ISSUE THREAD

**Issue by holdehn** [supabase/supabase#15312]
Describe the bug
I recently switched to an M2 Mac and since then, the Supabase CLI has stopped working for me. I'm encountering an error message that reads: "assertion failed [block != nullptr]: BasicBlock requested for unrecognized address (BuilderBase.h:550 block_for_offset)". The error occurs when I'm trying to use the CLI for various tasks.

To Reproduce
I'm not sure how to reproduce this error on other machines, but here are the steps I'm following on my machine:

Switch on my M2 Mac.
Open terminal.
Run any Supabase CLI command. For example, supabase start.
See the error message described above.
Expected behavior
I expect the Supabase CLI commands to execute without any errors, as they did before I switched to my M2 Mac.

System information
OS: macOS (M2 chip)
Version of supabase-js: (please fill in the version you're using)
Version of Node.js: v20.3.1 (arm64)
Additional context
I have checked that I'm running the arm64 version of Node.js, as I thought this could be a problem related to running x86 programs on an M1 Mac. However, the problem persists even though I'm using the correct architecture for my machine. I have found similar issues reported by other users, but their solutions have not resolved my problem.

I'm creating this issue to seek help with this problem and hopefully, this will also assist others who may encounter this error in the future.

**Comment by gregnr** [comment]
Hey @holdehn, very strange! Based on some quick research, it definitely seems to be a problem related to x86 emulation.

I know you mentioned updating to the `arm64` version of node, but would you mind double checking that was successful by running some of the debug steps outlined here:
https://github.com/ethereum/solidity/issues/13523#issuecomment-1544599434

Additionally would you mind sharing your `supabase` CLI version? If installed through `brew` you can run:
```shell
brew info supabase
```

We could also try double checking the arch for your `supabase` and `node` binaries:
```shell
file $(which supabase)
```
```shell
file $(which node)
```

Worth mentioning I was able to successfully run the supabase CLI on an M2. Hopefully we can sort out what is happening in your scenario.

**Comment by holdehn** [comment]
node versions:
`holden@MacBook-Air ~ % node -e 'console.log(process.arch)'
arm64
holden@MacBook-Air ~ % arch
arm64
`

Supabase cli version:

`holden@MacBook-Air ~ % brew info supabase

==> supabase/tap/supabase: stable 1.68.6
Supabase CLI
https://supabase.com
/opt/homebrew/Cellar/supabase/1.68.6 (8 files, 35.4MB) *
  Built from source on 2023-06-23 at 08:59:49
From: https://github.com/supabase/homebrew-tap/blob/HEAD/supabase.rb
License: MIT
==> Caveats
zsh completions have been installed to:
  /opt/homebrew/share/zsh/site-functions`


`holden@MacBook-Air ~ % file $(which supabase)
/opt/homebrew/bin/supabase: Mach-O 64-bit executable arm64
holden@MacBook-Air ~ % 
`

`holden@MacBook-Air ~ % file $(which node)

/Users/holden/.nvm/versions/node/v20.3.1/bin/node: Mach-O 64-bit executable arm64

**Comment by holdehn** [comment]
bump

really need to get this working :/

**Comment by holdehn** [comment]
@gregnr any update or suggestion on how to debug this?

**Comment by holdehn** [comment]
well it works now, i didn't actually do anything but :D

**Comment by holdehn** [comment]
still same error, it only worked one time @gregnr

**Comment by holdehn** [comment]
For some reason it allows me to create new function but does not allow me to deploy (this screenshot uses node installed cli but it also works with brew installed supabase, I can create new funcitons but not deploy

<img width="718" alt="Screenshot 2023-06-29 at 12 46 42 PM" src="https://github.com/supabase/supabase/assets/109185265/dedbf2a5-7b3b-42ad-98aa-f238d1badd1f">

**Comment by gregnr** [comment]
Hey @holdehn, that's a bummer. Confirming you've tried all the suggestions in this thread?
https://github.com/ethereum/solidity/issues/13523#issuecomment-1544599434

(uninstall node, install `nvm`, etc). Are you using VS Code? If so, worth checking if you get the same errors in a regular terminal.

**Comment by holdehn** [comment]
yup...i've tried everything in that thread and more. tried in both vs code and regular terminal. super disappointing because the computer is so fast and edge functions are a huge part of my work. any other suggestions to debug?

**Comment by filip-nikacevic** [comment]
@holdehn I had the same issue on my M1 Mac and was only able to resolve it by updating the supabase CLI to v 1.75.5 (which includes https://github.com/supabase/cli/pull/1280) as well as applying the fix of removing the ~/.supabase/deno folder from 
https://github.com/supabase/cli/issues/1266#issuecomment-1614910643.

Hopefully this helps!

**Comment by kiwicopple** [comment]
Closing the issue as a part of large repository cleanup, due to it being inactive and/or outdated. There's a large chance that this issue has been solved since.
Please do not hesitate to ping me if it is still relevant, and I will happily reopen and it.
Cheers!
